test/jdk/javax/swing/JFileChooser/4150029/bug4150029.java

```diff
@@ -21,73 +21,129 @@
+import java.io.File;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
 
 import jdk.test.lib.Platform;
 
-import javax.swing.*;
-import java.io.File;
-import java.io.IOException;
+/*
+ * @test
+ * @bug 4150029 8006087
+ * @summary BackSpace keyboard button does not lead to parent directory
+ * @library /test/lib /java/awt/regtesthelpers
+ * @build jdk.test.lib.Platform PassFailJFrame
+ * @run main/manual bug4150029
+ */
+
+public class bug4150029 {
+    private static JFrame frame;
+    private static JFileChooser fileChooser;
+    private static Robot robot;
+    private static File testDir;
+    private static File subDir;
+    private static File prevDir;
+    private static File crntDir;
 
-public class bug4150029 extends JApplet {
-    private boolean res;
+    public static void main(String[] args) throws Exception {
+        robot = new Robot();
+        robot.setAutoDelay(100);
 
-    public void init() {
-        if (Platform.isOSX()) {
-            try {
-                UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
-            } catch (Exception e) {
-                throw new RuntimeException(e);
+        try {
+            if (Platform.isOSX()) {
+                try {
+                    UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
             }
-        }
 
-        String tmpDir = System.getProperty("java.io.tmpdir");
+            String tmpDir = System.getProperty("java.io.tmpdir");
 
-        if (tmpDir.length() == 0) {//'java.io.tmpdir' isn't guaranteed to be defined
-            tmpDir = System.getProperty("user.home");
+            //'java.io.tmpdir' isn't guaranteed to be defined
+            if (tmpDir.length() == 0) {
+                tmpDir = System.getProperty("user.home");
+            }
+            System.out.println("Temp directory: " + tmpDir);
+
+            testDir = new File(tmpDir, "testDir");
+            testDir.mkdir();
+            testDir.deleteOnExit();
+            System.out.println("Created directory: " + testDir);
+
+            subDir = new File(testDir, "subDir");
+            subDir.mkdir();
+            subDir.deleteOnExit();
+            System.out.println("Created sub-directory: " + subDir);
+
+            fileChooser = new JFileChooser(subDir);
+
+//            fileChooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+
+            SwingUtilities.invokeAndWait(() -> {
+                frame = new JFrame("Backspace Shortcut for Directory Navigation Test");
+                frame.getContentPane().setLayout(new BorderLayout());
+                fileChooser = new JFileChooser(subDir);
+                fileChooser.setControlButtonsAreShown(false);
+                frame.getContentPane().add(fileChooser, BorderLayout.CENTER);
```

Comment:
```suggestion
                frame.add(fileChooser, BorderLayout.CENTER);
```

It was brought to my attention in one of my PRs that `getContentPane()` is obsolete as of JDK 1.5. They referenced  https://github.com/openjdk/jdk/pull/18390#discussion_r1533017080.